scsi: XSI-1728 Avoid repeatedly emitting uevents for LUN changes

SDEV_EVT_LUN_CHANGE_REPORTED events should only be reported when we're
not expecting the LUN change. Lest it may trigger udev rules which in
turn trigger more LUN change events.

Signed-off-by: Gerald Elder-Vass <gerald.elder-vass@cloud.com>
diff --git a/drivers/scsi/scsi_error.c b/drivers/scsi/scsi_error.c
index b7a8fdfeb2f4..4e0274607dae 100644
--- a/drivers/scsi/scsi_error.c
+++ b/drivers/scsi/scsi_error.c
@@ -418,7 +418,8 @@ static void scsi_report_sense(struct scsi_device *sdev,
 			sdev_printk(KERN_WARNING, sdev,
 				    "Inquiry data has changed");
 		} else if (sshdr->asc == 0x3f && sshdr->ascq == 0x0e) {
-			evt_type = SDEV_EVT_LUN_CHANGE_REPORTED;
+			if (!sdev->sdev_target->expecting_lun_change)
+				evt_type = SDEV_EVT_LUN_CHANGE_REPORTED;
 			scsi_report_lun_change(sdev);
 			sdev_printk(KERN_WARNING, sdev,
 				    "Warning! Received an indication that the "
